packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1274 @@
+/// also be closed by swiping downwards.
+///
+/// The [previewBuilder] parameter is most commonly used to display a slight
+/// variation of [child]. See [previewBuilder] for an example of rounding the
+/// child's corners and allowing its aspect ratio to expand, similar to the
+/// Photos app on iOS.
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This sample shows a very simple CupertinoContextMenu for an empty red
+/// 100x100 Container. Simply long press on it to open.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     key: scaffoldKey,
+///     body: Center(
+///       child: Container(
+///         width: 100,
+///         height: 100,
+///         child: CupertinoContextMenu(
+///           child: Container(
+///             color: Colors.red,
+///           ),
+///           actions: <CupertinoContextMenuAction>[
+///             CupertinoContextMenuAction(
+///               child: const Text('Action one'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///             CupertinoContextMenuAction(
+///               child: const Text('Action two'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///           ],
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [Apple's HIG for Context Menus](https://developer.apple.com/design/human-interface-guidelines/ios/controls/context-menus/)
+class CupertinoContextMenu extends StatefulWidget {
+  /// Create a context menu.
+  ///
+  /// [actions] is required and cannot be null. It must contain at least one
+  /// [CupertinoContextMenuAction].
+  ///
+  /// [child] is required and cannot be null.
+  CupertinoContextMenu({
+    Key key,
+    @required this.actions,
+    @required this.child,
+    this.previewBuilder,
+  }) : assert(actions != null && actions.isNotEmpty),
+       assert(child != null),
+       super(key: key);
+
+  /// The widget that can be "opened" with the [CupertinoContextMenu].
+  ///
+  /// When the [CupertinoContextMenu] is long-pressed, the menu will open and
+  /// this widget (or the widget returned by [previewBuilder], if provided) will
+  /// be moved to the [Overlay] and placed inside of an [Expanded] widget. This
+  /// allows the child to resize to fit in the full-screen Overlay, if it
+  /// doesn't size itself.
+  ///
+  /// When the [CupertinoContextMenu] is "closed", this widget acts like a
+  /// [Container], i.e. it does not constrain its child's size or affect its
+  /// position.
+  ///
+  /// This parameter cannot be null.
+  final Widget child;
+
+  /// The actions that are shown in the menu.
+  ///
+  /// This parameter cannot be null, and items in the List must be
+  /// [CupertinoContextMenuAction]s.
+  final List<CupertinoContextMenuAction> actions;
```

Comment:
Yeah there is nothing preventing us from using `List<Widget>` here.  I did it just to encourage users to always use CupertinoContextMenuActions, but I also didn't realize there were group separators.  Does Flutter usually allow any type of widget when it can?

Reply:
There's an issue for this: https://github.com/flutter/flutter/issues/12078. Also I tried adding a context menu to a `UISegmentedControl`, and it turns out other than group separators there can also be headers (similar to action sheet headers)

Reply:
If that's more in line with how we usually do it then changing it sounds good to me.

Reply:
Oh sorry I posted my last comment before Github updated to show yours.  That issue definitely confirms that this should be agnostic so I'm changing it now :+1: